File: 30daysOFcode_Python/day 14.py
class Difference:
    def __init__(self, a):
        self.__elements = a

        # Add your code heredef computeDifference(self):
    def computeDifference(self):
        self.maximumDifference = abs(
            max(self.__elements) - min(self.__elements))

# End of Difference class


_ = input()
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)

File: 30daysOFcode_Swift/day 14.swift
import Foundation

class Difference {
    private var elements = [Int]()
    var maximumDifference: Int

    init(a: [Int]) {
        elements = a
        maximumDifference = 0
    }

    func computeDifference() {
        maximumDifference = elements.reduce(0) { result, element in
            max(result, elements.map { abs(element - $0) }.max() ?? 0)
        }
    }
} // End of Difference class

let n = Int(readLine()!)!
let a = readLine()!.components(separatedBy: " ").map{ Int($0)! }

let d = Difference(a: a)

d.computeDifference()

print(d.maximumDifference)
